lib/Ohmbrewer_STILL.cpp

```diff
@@ -0,0 +1,558 @@
+
+    // Print the label
+    screen->resetTextColor();
+    screen->print(" C. Pump: "); // We want a little margin
+
+    // Print the state
+    if (getCoolantPump()->getState()){
+        screen->setTextColor(screen->YELLOW, screen->DEFAULT_BG_COLOR);
+        screen->print("ON ");
+    } else {
+        screen->setTextColor(screen->RED, screen->DEFAULT_BG_COLOR);
+        screen->print("OFF");
+    }
+
+    screen->printMargin(2);
+    screen->resetTextColor();
+
+    return micros() - start;
+}
+
+/**
+ * Publishes updates to Ohmbrewer, etc.
+ * This function is called by update().
+ * @param args The argument string passed into the Particle Cloud
+ * @param argsMap A map representing the key/value pairs for the update
+ * @returns The time taken to run the method
+ */
+int Ohmbrewer::STILL::doUpdate(String &args, Ohmbrewer::Equipment::args_map_t &argsMap) {
+    unsigned long start = millis();
+
+    // If there are any remaining parameters
+    if(args.length() > 0) {
+        String tunSensorKey = String("kettle_sensor_state");
+        String rPumpKey = String("c_pump_state");
+        String tubeKey = String("therm_params");
+
+        parseArgs(args, argsMap);
+
+        //TODO flush out with all vars
+        // The remaining settings are optional/convenience parameters
+        if(argsMap.count(tunSensorKey) != 0) {
+            if(argsMap[tunSensorKey].equalsIgnoreCase("ON")) {
+                getKettleSensor()->setState(true);
+            } else if(argsMap[tunSensorKey].equalsIgnoreCase("OFF")) {
+                getKettleSensor()->setState(false);
+            } else if(argsMap[tunSensorKey].equalsIgnoreCase("--")) {
+                // Do nothing. Intentional.
+            } else {
+                // Do nothing. TODO: Should probably raise an error code...
+            }
+        }
+
+        if(argsMap.count(rPumpKey) != 0) {
+            if(argsMap[rPumpKey].equalsIgnoreCase("ON")) {
+                getCoolantPump()->setState(true);
+            } else if(argsMap[rPumpKey].equalsIgnoreCase("OFF")) {
+                getCoolantPump()->setState(false);
+            } else if(argsMap[rPumpKey].equalsIgnoreCase("--")) {
+                // Do nothing. Intentional.
+            } else {
+                // Do nothing. TODO: Should probably raise an error code...
+            }
+        }
+
+
+        if(argsMap.count(tubeKey) != 0) {
+            // Pass the remaining parameters down to the Tube (a Thermostat)
+            getTherm()->doUpdate(argsMap[tubeKey], argsMap);
+        }
+
+    }
+
+
+    return millis() - start;
+}
+
+/**
+ * Reports which of the Rhizome's pins are occupied by the
+ * Equipment, forming a logical Sprout.
+ * @param pins The list of physical pins that the Equipment is connected to.
+ */
+void Ohmbrewer::STILL::whichPins(std::list<int>* pins){
+
+    pins->push_back(_kettleSensor->getBusPin()); //assumes onewire protocol
+    _therm->whichPins(pins);
+    _coolant->whichPins(pins);
```

Comment:
You need to push_back the results for all of these (2 out of 3 aren't getting pushed in)